.github/workflows/integration.yml

```diff
@@ -16,20 +16,13 @@ jobs:
         python-version: ["3.11"]
 
     steps:
-      - uses: actions/checkout@v2
-
-      - name: pip cache
-        uses: actions/cache@v2
-        with:
-          path: ~/.cache/pip
-          key: integration-pip-${{ hashFiles('**/pyproject.toml') }}
-          restore-keys: |
-            integration-pip-
+      - uses: actions/checkout@v4
 
       - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v2
+        uses: actions/setup-python@v5
         with:
           python-version: ${{ matrix.python-version }}
+          cache: "pip"
```

Comment:
Shouldn't this be `poetry`?

Reply:
[The docs for the `actions/setup-python` caching.](https://github.com/actions/setup-python?tab=readme-ov-file#caching-packages-dependencies)